CMakeLists.txt

```diff
@@ -0,0 +1,68 @@
+cmake_minimum_required(VERSION 3.9)
+
+# Fail immediately if not using an out-of-source build
```

Comment:
Just curious but what are the actual issues with building *in* the source? I mean, I generally *prefer* out-of-source builds myself but thats a bit different than wholly disallowing them in source.

Reply:
When you run CMake it creates all kind of Makefiles. In the case of in building in source it will overwrite existing Makefile at the top level. I want to make sure that does not happen since I would like to preserve your existing build system.